AR-3662

imtithalhalalo opened the pull request:
item / sales tab - datagrid
change the table to become a datagrid

NoraBitarSM commented:
since you are not using the SalesForm anymore in this form, please check if it's not used somewhere else and delete it

Ghinwa-Arab commented:
<img width="1601" height="897" alt="image" src="https://github.com/user-attachments/assets/1669645a-67d9-40bd-9e2a-e373f8f4270c" />
"price with vat" column is missing, on purpose?

Ghinwa-Arab commented:
<img width="1377" height="738" alt="image" src="https://github.com/user-attachments/assets/238e9c5a-db23-4b50-83d0-7b6890a02829" />
If none of the mandatory fields have values, treat the column as optional so that saving is still allowed

Ghinwa-Arab commented:
![WhatsApp Image 2025-08-22 at 08 07 51_cc680c18](https://github.com/user-attachments/assets/86981aaf-6270-462a-bd64-4782dd666cf8)
For item "IG-00018", there was no data in the Sales tab. I added the data as shown in the screenshot, saved it, and then closed the window. When I tried to reopen it, the request "SA.asmx/qryPR?&_itemId=312&_currencyId=0" returned an error. Please ensure that the data is being saved and sent correctly.

imtithalhalalo commented:
> <img alt="image" width="1290" height="751" src="https://private-user-images.githubusercontent.com/116278589/480811960-a151243c-285a-434d-91ac-e3b43cb2c129.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU4NDQwNTMsIm5iZiI6MTc1NTg0Mzc1MywicGF0aCI6Ii8xMTYyNzg1ODkvNDgwODExOTYwLWExNTEyNDNjLTI4NWEtNDM0ZC05MWFjLWUzYjQzY2IyYzEyOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyMlQwNjIyMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xZTBkZDRhMzlhYWRhM2UxMDI0ZTI0YmFkZDAxNjE3OTk5YWJlNWZlYmJhZGM2N2RjOTQ0MjQ5MWNkZDhmOTk0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LylntWxkpXNNmLf83SxFULKhIISZqxUcwTXEi1JiYU0">
> Also for this record, I deleted the middle row and added a new one at the end. When reopening the window, it’s returning an error as well.

api issue fixed

imtithalhalalo commented:
> ![WhatsApp Image 2025-08-22 at 08 07 51_cc680c18](https://private-user-images.githubusercontent.com/116278589/480810335-86981aaf-6270-462a-bd64-4782dd666cf8.jpg?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTU4NDQwNTMsIm5iZiI6MTc1NTg0Mzc1MywicGF0aCI6Ii8xMTYyNzg1ODkvNDgwODEwMzM1LTg2OTgxYWFmLTYyNzAtNDYyYS1iZDY0LTQ3ODJkZDY2NmNmOC5qcGc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwODIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDgyMlQwNjIyMzNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iM2ZiMzc2MzFkMzZmNTFiMzAxYjU4ODg3NzcxZjIxMWM4NGY4MDFhZDFjNzM5NGZiMmFmMTM5MGIwZWE4ZmMxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.8R47m0lbjG81pXh0Qsd51TeT3lZEiBTOuNchuRGretA) For item "IG-00018", there was no data in the Sales tab. I added the data as shown in the screenshot, saved it, and then closed the window. When I tried to reopen it, the request "SA.asmx/qryPR?&_itemId=312&_currencyId=0" returned an error. Please ensure that the data is being saved and sent correctly.

api issue fixed

Ghinwa-Arab commented:
<img width="624" height="190" alt="image" src="https://github.com/user-attachments/assets/a18b0e27-dea2-4388-87d1-89df21061c75" />

shouldn't accept negative nbrs

Ghinwa-Arab commented:
<img width="1311" height="410" alt="image" src="https://github.com/user-attachments/assets/97bd7419-0a71-4bf4-b42c-d530bfdea87e" />
 min price is not mandatory

Ghinwa-Arab commented:
<img width="1129" height="878" alt="image" src="https://github.com/user-attachments/assets/ca457eae-e0f3-4998-afd1-ac441d9f001e" />
```

curl 'https://deploy.arguserp.net/IV.asmx/getMFR?_recordId=312' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhY2MiOiIxMDUiLCJ1c3IiOiI4NSIsImRiZSI6IjEiLCJkYnMiOiIyIiwiZG93bnRpbWVSZXNvdXJjZSI6ImZhbHNlIiwibmJmIjoxNzU2MTA0MTMwLCJleHAiOjE3NTYxMzI5MzAsImlhdCI6MTc1NjEwNDEzMH0.OdrloZal4e-CZMBJBt9SRiXDcNytu5xYnhodfYAVTmU' \
  -H 'cache-control: no-cache' \
  -H 'languageid: 1' \
  -H 'origin: https://argus-fe-git-ar-3662-george-kalashs-projects.vercel.app' \
  -H 'pragma: no-cache' \
  -H 'priority: u=1, i' \
  -H 'referer: https://argus-fe-git-ar-3662-george-kalashs-projects.vercel.app/' \
  -H 'sec-ch-ua: "Not;A=Brand";v="99", "Microsoft Edge";v="139", "Chromium";v="139"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Windows"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: cross-site' \
  -H 'user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/139.0.0.0 Safari/537.36 Edg/139.0.0.0'
```

Ghinwa-Arab commented:
<img width="1117" height="782" alt="image" src="https://github.com/user-attachments/assets/c8f5277d-6ac2-42d3-be78-5436e291e67a" />

on Save i'm getting this error

```
curl 'https://deploy.arguserp.net/IV.asmx/setIT' \
  -H 'accept: application/json, text/plain, */*' \
  -H 'accept-language: en-US,en;q=0.9' \
  -H 'authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhY2MiOiIxMDUiLCJ1c3IiOiI4NSIsImRiZSI6IjEiLCJkYnMiOiIyIiwiZG93bnRpbWVSZXNvdXJjZSI6ImZhbHNlIiwibmJmIjoxNzU2MTA0MTMwLCJleHAiOjE3NTYxMzI5MzAsImlhdCI6MTc1NjEwNDEzMH0.OdrloZal4e-CZMBJBt9SRiXDcNytu5xYnhodfYAVTmU' \
  -H 'cache-control: no-cache' \
  -H 'content-type: multipart/form-data; boundary=----WebKitFormBoundaryOMQGQQQhfVmBbOrI' \
  -H 'languageid: 1' \
  -H 'origin: https://argus-fe-git-ar-3662-george-kalashs-projects.vercel.app' \
  -H 'pragma: no-cache' \
  -H 'priority: u=1, i' \
  -H 'referer: https://argus-fe-git-ar-3662-george-kalashs-projects.vercel.app/' \
  -H 'sec-ch-ua: "Not;A=Brand";v="99", "Microsoft Edge";v="139", "Chromium";v="139"' \
  -H 'sec-ch-ua-mobile: ?0' \
  -H 'sec-ch-ua-platform: "Windows"' \
  -H 'sec-fetch-dest: empty' \
  -H 'sec-fetch-mode: cors' \
  -H 'sec-fetch-site: cross-site' \
  -H 'user-agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/139.0.0.0 Safari/537.36 Edg/139.0.0.0' \
  --data-raw $'------WebKitFormBoundaryOMQGQQQhfVmBbOrI\r\nContent-Disposition: form-data; name="record"\r\n\r\n{"spfName":"default","categoryName":"GOLD 21","lotCategoryName":null,"productionClassName":null,"msName":"Weight ","ptName":"Unit Quantity","categoryRef":"G21","weight":14481.61,"volume":779.4444,"labelTemplateId":null,"collectionName":"234567898765432456789098765432","collectionRef":"1234567876","sku":"IG-00018","shortName":"312","flName":"312","description":"3213231","categoryId":61,"msId":3,"defSaleMUId":"","groupId":25,"spfId":7,"trackBy":1,"lotCategoryId":null,"pgId":3,"taxId":1,"returnPolicyId":29,"priceType":1,"procurementMethod":1,"valuationMethod":2,"ivtItem":true,"salesItem":true,"purchaseItem":true,"kitItem":false,"isInactive":false,"productionLevel":4,"collectionId":5,"unitPrice":null,"name":"Gold Earrings","recordId":312}\r\n------WebKitFormBoundaryOMQGQQQhfVmBbOrI--\r\n'
```

Ghinwa-Arab commented on src/pages/iv-items/forms/SaleList.js:
```diff
@@ -47,134 +69,187 @@ const SalesList = ({ store, labels, maxAccess, formikInitial }) => {
-      refetch()
-    })()
-  }, [formik.values.currencyId])
+    formik.setFieldValue('defSaleMUId', store.measurementId || '')
```
for all add || null instead ''

Ghinwa-Arab commented:
<img width="1293" height="229" alt="image" src="https://github.com/user-attachments/assets/e3694a7b-e1f6-464f-b58a-3b626d44a087" />
i should be able to clear the grid

Ghinwa-Arab commented:
<img width="1325" height="767" alt="image" src="https://github.com/user-attachments/assets/6d6d13b4-d4d9-4b07-bbca-e229fe53e23f" />
